zproject/backends.py

```diff
@@ -585,6 +593,31 @@ def email_belongs_to_ldap(realm: Realm, email: str) -> bool:
         return False
 
 
+def ensure_user_groups_for_ldap_sync_exist() -> None:
+    """
+    Ensures that the UserGroup objects set up for syncing with LDAP
+    in LDAP_GROUP_SYNC_LIST exist.
+    """
+
+    for subdomain, ldap_group_names_for_sync in settings.LDAP_GROUP_SYNC_LIST.items():
+        realm = Realm.objects.get(string_id=subdomain)
+        existing_group_names = UserGroup.objects.filter(
+            realm=realm, name__in=ldap_group_names_for_sync
+        ).values_list("name", flat=True)
+
+        for group_name in ldap_group_names_for_sync:
+            if group_name in existing_group_names:
+                continue
+
+            ldap_logger.debug(
+                "creating zulip group %s from ldap for realm %s", group_name, subdomain
+            )
+            user_group = create_user_group_in_database(
+                group_name, [], realm, acting_user=None, description="Created by LDAP sync"
+            )
+            do_send_create_user_group_event(user_group, [])
```

Comment:
@sahil839 I'm concerned that this function might not be a good from the standpoint that if today or in the future, we have any mandatory configuration settings that need to be done when completing a new user group (and I think our plans for making groups do a lot more may include that), then there won't be a source we can use to specify the properties to use when creating those groups.

@ClundXIII can you talk about the motivation for having groups be automatically created in this sync code path, vs. created in the UI?

Reply:
I guess we would have default values for all the settings.

Reply:
> @ClundXIII can you talk about the motivation for having groups be automatically created in this sync code path, vs. created in the UI?

We have been running this setup with a few additional patches and additional half-baked automations (the group sync over [here](https://github.com/zulip/zulip/pull/24490)). The bottom line is that there might be a bunch of new groups created at once and this way we only change the mapping and the groups get created automatically. The goal is to mirror Rocketchats group sync feature. (So there is at least one follow up PR after this one.)

Removing this auto creation is up to you. I imagine that this patch overall will be replaced by a UI configuration at some point anyway. Thats fine with us.

Reply:
OK, let's remove it, I think that'll be a lot more future-proof.